release

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env -S bash -e
+
+cd $(dirname $0)
+
+main() {
+  local version="$1"
+  local preview_txid="$2"
+  local preprod_txid="$3"
+
+  [[ -z "$version" ]]      && usage "Missing version number"
+  [[ -z "$preview_txid" ]] && usage "Missing preview network hydra scripts publication transaction id"
+  [[ -z "$preprod_txid" ]] && usage "Missing preprod network hydra scripts publication transaction id"
+
+  check_can_release "$version"
+
+  prepare_release "$version" "$preview_txid" "$preprod_txid"
+
+  publish_release # fake for now
+}
+
+usage() {
+  local message=$1
+
+  cat >&2 <<EOF
+$message
+
+  $0 <version> <preview_txid> <preprod_txid>
+Publishes a new release of hydra.
+<version> must respect [Semantic Versioning](http://semver.org/)
+<preview_txid> is the id of the transaction used to publish the hydra scripts on the preview network (see smoke tests)
+<preprod_txid> is the id of the transaction used to publish the hydra scripts on the preprod network (see smoke tests)
+EOF
+
+  exit 1
+}
+
+check_can_release() {
+  local version="$1"
+
+  check_no_uncommited_changes  
+  check_version_is_valid $version
+  check_changelog_is_up_to_date $version
+
+  true #avoid error on last instruction of function (see bash -e)
+}
+
+prepare_release() {
+  local version="$1"
+  local preview_txid="$2"
+  local preprod_txid="$3"
+
+  update_cabal_version "$version"
+  update_api_version   "$version"
+  update_preview_txid "$version" "$preview_txid"
+  update_preprod_txid "$version" "$preprod_txid"
+
+  find . -name '*-e' -exec rm '{}' \; # cleanup BSD sed mess
```

Comment:
What is the BSD sed mess?

Reply:
BSD sed generates backup files when you do inline substitution. These files are names `<original file name>-e`.

For instance line [129](https://github.com/input-output-hk/hydra/blob/9280174ff9c6ae978ed3b2439c56cdb0157cf442/release#L129) here would generate a backup file named `hydra-node/json-schemas/api.yaml-e` with BSD sed and would not with GNU sed. And guess what, you would use BSD sed on OS X and GNU sed on linux ;)

Anyway, I've not been able to prevent sed on BSD to generate these files so I've added this `find` to cleanup the repo from them and avoid strange things when using BSD sed and that should not have any impact if one uses GNU sed.